motor_driver.c

```diff
@@ -0,0 +1,94 @@
+#include <stdio.h>
+
+#include "motor_driver.h"
+#include "nuts_bolts.h"
+#include "spi.h"
+
+#define ADDRESS_IDX 5U
+
+#define DECMOD_MASK     0x0F
+#define DECMOD_IDX      8U
+
+#define TORQUE_MASK      0xFF
+#define TORQUE_IDX       0
+
+#define ENABLE_MASK     0x1
+#define ENABLE_IDX      0
+
+#define STEPS_MASK      0x0F
+#define STEPS_IDX       3U
+
+void _motor_drv_write_reg(enum stepper_e stepper, enum address_e address, uint8_t * data)
+{
+  uint8_t data_out[2];
+  data_out[0] = (address << ADDRESS_IDX) | (data[1] & 0x0F);
+  data_out[1] = data[0];
+
+  bit_true(SCS_PORT, 1 << scs_pin_lookup[stepper]);
+  spi_write(data_out, 2);
+  bit_false(SCS_PORT, 1 << scs_pin_lookup[stepper]);
+
+}
+
+void _motor_drv_read_reg(enum stepper_e stepper,
+                         enum address_e address,
+                         uint8_t * data)
+{
+  uint8_t data_out[2];
+  data_out[0] = address << ADDRESS_IDX;
+  data_out[1] = 0;
```

Comment:
You can simplify this as:

```
uint8_t data_out[2] = {address << ADDRESS_IDX, 0};
```

(Tested with normal gcc, should work here as well)